src/required_resources.h

```diff
@@ -19,9 +19,39 @@ namespace bpftrace {
 
 class BPFtrace;
 
-struct HelperErrorInfo {
-  int func_id = -1;
-  ast::Location loc;
+class HelperErrorInfo {
+public:
+  // This classe effectively wraps a location, but preserves only the parts
+  // that are needed to emit the error in a useful way. This is because it may
+  // be serialized and used by a separate runtime.
```

Comment:
wait what separate runtime?

Reply:
AOT

Reply:
That's what I thought - maybe good to be specific but NBD.

Reply:
I try to avoid being too prescriptive, the important bits are that it is an isolated unit which is serialized. (As is the rest of the resources in this file.)

Reply:
Obviously non-blocking but since it seems you like discussing the color of the bike shed 😄  I think being vague has the downside that developers reading this comment will have the exact same question in their head as I did and right now there isn't a separate runtime aside from AOT, which is barely supported at this point. So I don't find it so much "prescriptive" as "accurate" and "clarifying".